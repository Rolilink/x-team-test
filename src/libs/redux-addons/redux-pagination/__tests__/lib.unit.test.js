import withPagination, { paginationFactory, cases } from '../';

const { setPage, setLimit, incrementPage, decrementPage } = cases;

describe('cases.setPage', () => {
  it('should set page from payload', () => {
    const state = {
      items: [],
      pagination: {
        page: 0,
        limit: 10,
      },
    };

    const action = { payload: { pagination: { page: 5 } } };

    expect(setPage(state, action)).toMatchSnapshot();
  });
});

describe('cases.setLimit', () => {
  it('should set limit from payload', () => {
    const state = {
      items: [],
      pagination: {
        page: 0,
        limit: 10,
      },
    };

    const action = { payload: { pagination: { limit: 5 } } };

    expect(setLimit(state, action)).toMatchSnapshot();
  });
});

describe('cases.incrementPage', () => {
  it('should increment page from state', () => {
    const state = {
      items: [],
      pagination: {
        page: 0,
        limit: 10,
      },
    };

    expect(incrementPage(state)).toMatchSnapshot();
  });
});

describe('cases.decrementPage', () => {
  it('should decrement page from state', () => {
    const state = {
      items: [],
      pagination: {
        page: 2,
        limit: 10,
      },
    };

    expect(decrementPage(state)).toMatchSnapshot();
  });

  it('should never decrement less than zero', () => {
    const state = {
      items: [],
      pagination: {
        page: 0,
        limit: 10,
      },
    };

    expect(decrementPage(state)).toMatchSnapshot();
  });
});

describe('paginationFactory', () => {
  it('should return default state', () => {
    expect(paginationFactory()).toMatchSnapshot();
  });
});

describe('enhancer.withPagination', () => {
  it('should enhance received state to add network state', () => {
    const state = {
      items: [],
    };

    expect(withPagination(state)).toMatchSnapshot();
  });

  it('should be able to initialize network state', () => {
    const state = {
      items: [],
    };
    const initialPaginationState = {
      page: 1,
      limit: 10,
    };

    expect(withPagination(state, initialPaginationState)).toMatchSnapshot();
  });
});
