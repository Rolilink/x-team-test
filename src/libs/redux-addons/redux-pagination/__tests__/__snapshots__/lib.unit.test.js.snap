// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cases.decrementPage should decrement page from state 1`] = `
Object {
  "items": Array [],
  "pagination": Object {
    "limit": 10,
    "page": 1,
  },
}
`;

exports[`cases.decrementPage should never decrement less than zero 1`] = `
Object {
  "items": Array [],
  "pagination": Object {
    "limit": 10,
    "page": 0,
  },
}
`;

exports[`cases.incrementPage should increment page from state 1`] = `
Object {
  "items": Array [],
  "pagination": Object {
    "limit": 0,
    "page": 1,
  },
}
`;

exports[`cases.setLimit should set limit from payload 1`] = `
Object {
  "items": Array [],
  "pagination": Object {
    "limit": 5,
    "page": 0,
  },
}
`;

exports[`cases.setPage should set page from payload 1`] = `
Object {
  "items": Array [],
  "pagination": Object {
    "limit": 10,
    "page": 5,
  },
}
`;

exports[`enhancer.withPagination should be able to initialize network state 1`] = `
Object {
  "items": Array [],
  "pagination": Object {
    "limit": 10,
    "page": 1,
  },
}
`;

exports[`enhancer.withPagination should enhance received state to add network state 1`] = `
Object {
  "items": Array [],
  "pagination": Object {
    "limit": 0,
    "page": 0,
  },
}
`;

exports[`paginationFactory should return default state 1`] = `
Object {
  "limit": 0,
  "page": 0,
}
`;
